File: Faltantes o por revisar/ArticulationBridges.py

from collections import defaultdict 

class Graph: 
	def __init__(self,vertices): 
		self.V= vertices 
		self.graph = defaultdict(list)
		self.Time = 0
	def addEdge(self,u,v): 
		self.graph[u].append(v) 
		self.graph[v].append(u) 
	def APUtil(self,u, visited, ap, parent, low, disc): 
		children =0
		visited[u]= True
		disc[u] = self.Time 
		low[u] = self.Time 
		self.Time += 1
		for v in self.graph[u]: 
			if visited[v] == False : 
				parent[v] = u 
				children += 1
				self.APUtil(v, visited, ap, parent, low, disc) 
				low[u] = min(low[u], low[v]) 
				if parent[u] == -1 and children > 1: 
					ap[u] = True
				if parent[u] != -1 and low[v] >= disc[u]: 
					ap[u] = True			 
			elif v != parent[u]: 
				low[u] = min(low[u], disc[v]) 
	def AP(self): 
		visited = [False] * (self.V) 
		disc = [float("Inf")] * (self.V) 
		low = [float("Inf")] * (self.V) 
		parent = [-1] * (self.V) 
		ap = [False] * (self.V) 
		for i in range(self.V): 
			if visited[i] == False: 
				self.APUtil(i, visited, ap, parent, low, disc) 
		for index, value in enumerate (ap): 
			if value == True: print index, 
g1 = Graph(5) 
g1.addEdge(1, 0) 
g1.addEdge(0, 2) 
g1.addEdge(2, 1) 
g1.addEdge(0, 3) 
g1.addEdge(3, 4) 
print "Puntos de articulacion en el grafo "
g1.AP() 
g2 = Graph(4) 
g2.addEdge(0, 1) 
g2.addEdge(1, 2) 
g2.addEdge(2, 3) 
print "Puntos de articulacion en el grafo "
g2.AP() 
g3 = Graph (7) 
g3.addEdge(0, 1) 
g3.addEdge(1, 2) 
g3.addEdge(2, 0) 
g3.addEdge(1, 3) 
g3.addEdge(1, 4) 
g3.addEdge(1, 6) 
g3.addEdge(3, 5) 
g3.addEdge(4, 5) 
print "Puntos de articulacion en el grafo "
g3.AP() 

File: Faltantes o por revisar/ArticulationBridges.cpp
#include <cstdlib>
#include <bits/stdc++.h>
#define  MAX 105
using namespace std;

struct Graph {
    int V;
    vector<int> adj[MAX];
    int time = 0;
    int NIL = -1;

    void addEdge(int v, int w) {
        adj[v].emplace_back(w);
        adj[w].emplace_back(v);
    }

    void APUtil(int u, bool visited[], int disc[],
            int low[], int parent[], bool ap[]) {
        static int time = 0;
        int children = 0;
        visited[u] = true;
        disc[u] = low[u] = ++time;
        for (int i = 0; i < adj[u].size(); i++) {
            int v = adj[u].at(i);
            if (!visited[v]) {
                children++;
                parent[v] = u;
                APUtil(v, visited, disc, low, parent, ap);
                low[u] = min(low[u], low[v]);
                if (parent[u] == NIL && children > 1)
                    ap[u] = true;
                if (parent[u] != NIL && low[v] >= disc[u])
                    ap[u] = true;
            } else if (v != parent[u])
                low[u] = min(low[u], disc[v]);
        }
    }

    void AP() {
        bool visited[V];
        int disc[V];
        int low [V];
        int parent[V];
        bool ap[V];
        for (int i = 0; i < V; i++) {
            parent[i] = NIL;
            visited[i] = false;
            ap[i] = false;
        }
        for (int i = 0; i < V; i++)
            if (visited[i] == false)
                APUtil(i, visited, disc, low, parent, ap);
        for (int i = 0; i < V; i++)
            if (ap[i] == true)
                cout << i << " ";
    }


};

int main() {
    // Create graphs given in above diagrams 
    cout << "Puntos de articulación en el grafo" << endl;
    Graph g1;
    g1.V = 5;
    g1.addEdge(1, 0);
    g1.addEdge(0, 2);
    g1.addEdge(2, 1);
    g1.addEdge(0, 3);
    g1.addEdge(3, 4);
    g1.AP();
    cout << endl;

    cout << "Puntos de articulación en el grafo" << endl;
    Graph g2;
    g2.V = 4;
    g2.addEdge(0, 1);
    g2.addEdge(1, 2);
    g2.addEdge(2, 3);
    g2.AP();
    cout << endl;

    cout << "Puntos de articulación en el grafo" << endl;
    Graph g3;
    g3.V=7;
    g3.addEdge(0, 1);
    g3.addEdge(1, 2);
    g3.addEdge(2, 0);
    g3.addEdge(1, 3);
    g3.addEdge(1, 4);
    g3.addEdge(1, 6);
    g3.addEdge(3, 5);
    g3.addEdge(4, 5);
    g3.AP();
}
